services/vocab_mapper/vocab_mapper.py

```diff
@@ -0,0 +1,159 @@
+                }),
+                "metadata": {},
+                "score": None
+            } for s in matches])
+        
+        return keyword_results + vector_results
+
+    def get_shortlist(self, input_text: str, general_info: str, specific_info: str, 
+                     expanded_terms: str, search_results: list) -> str:
+        """Step 3: Get a shortlist of the best matches."""
+        user_prompt = SHORTLIST_USER_PROMPT.format(
+            input_text=input_text,
+            general_info=general_info,
+            specific_info=specific_info,
+            expanded_terms=expanded_terms,
+            search_results=search_results
+        )
+        return self._call_llm(SHORTLIST_SYSTEM_PROMPT, user_prompt)
+
+    def get_final_selection(self, input_text: str, general_info: str, specific_info: str,
+                          expanded_terms: str, search_results: list) -> str:
+        """Step 4: Get the best match."""
+        user_prompt = FINAL_SELECTION_USER_PROMPT.format(
+            input_text=input_text,
+            general_info=general_info,
+            specific_info=specific_info,
+            expanded_terms=expanded_terms,
+            search_results=search_results
+        )
+        return self._call_llm(FINAL_SELECTION_SYSTEM_PROMPT, user_prompt)
+
+    def map_term(self, input_term: str, general_info: str = "", specific_info: str = "") -> dict:
+        """Map an input term using the target dataset."""
+        # Step 1: Get expanded terms
+        expanded_terms = self.get_expanded_terms(input_term, general_info, specific_info)
+        
+        # Step 2: Search database
+        search_results = self.search_database(expanded_terms)
+        
+        # Step 3: Get shortlist of best terms
+        shortlist = self.get_shortlist(
+            input_term, general_info, specific_info,
+            expanded_terms, search_results
+        )
+        
+        # Step 4: Select the best term (from the full search results)
+        final_selection = self.get_final_selection(
+            input_term, general_info, specific_info,
+            expanded_terms, search_results
+        )
+        
+        # Return all results
+        return {
+            "expanded_terms": expanded_terms,
+            # "search_results": search_results,  # large - can add back in for debugging
+            "shortlist": shortlist,
+            "final_selection": final_selection
+        }
+    
+
+def main(data):
+    """Map vocab with the VocabMapper using Google Sheets input data. Format output for Google Sheets."""
+    import os
+    from dotenv import load_dotenv
+    from embeddings import loinc_store
+    from datasets import load_dataset
+
+    # Format Google Sheets data
+    input_data, column_indices, original_values = format_google_sheets_input(data)
+
+    # Initialize and setup mapper
+    load_dotenv(override=True)
+    ANTHROPIC_API_KEY = os.getenv("ANTHROPIC_API_KEY")
```

Comment:
Note: we should read the api key from the incoming args object first, and if it's not there, default it from the env. Refer to `job_chat` for an example